test/cypress/integration/simplified-account/service-settings/api-keys/api-keys.cy.js

```diff
@@ -166,6 +167,45 @@ describe('Settings - API keys', () => {
         })
       })
     })
+
+    describe('re-name an api key', () => {
+      const NEW_API_KEY_NAME = 'api key description' // pragma: allowlist secret
+      const TOKEN_LINK = 'token-link-1'
+
+      const apiKeys = [
+        new Token().withCreatedBy('algae bra').withDescription('mathematical clothes')
+          .withIssuedDate('10 Dec 2024').withLastUsed('10 Dec 2024').withTokenLink(TOKEN_LINK)
+      ]
+
+      beforeEach(() => {
+        setupStubs('admin', apiKeys)
+        cy.task('setupStubs', [
+          apiKeysStubs.changeApiKeyName(TOKEN_LINK, NEW_API_KEY_NAME)
+        ])
+        cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/api-keys`)
+      })
+
+      it('show the API key name page', () => {
+        cy.get('.govuk-summary-card').within(() => {
+          cy.contains('h2', 'mathematical clothes').should('exist')
+          cy.contains('a', 'Change name').click()
+        })
+        cy.contains('h1', 'API key name').should('exist')
+      })
+
+      it('should re-name the api key successfully', () => {
+        cy.get('.govuk-summary-card').within(() => {
+          cy.contains('h2', 'mathematical clothes').should('exist')
+          cy.contains('a', 'Change name').click()
+        })
+        cy.get('input[id="description"]').type(NEW_API_KEY_NAME)
+        cy.contains('button', 'Continue').click()
+        cy.contains('h1', 'Test API keys').should('exist')
+        cy.get('.govuk-summary-card').within(() => {
+          cy.contains('h2', 'mathematical clothes').should('exist')
```

Comment:
shouldn't this change to the new name?

Reply:
Ohh you're right. I'm not sure this test line is relevant because we're stubbing out the response [here](https://github.com/alphagov/pay-selfservice/pull/4399/files#diff-62c503bb6b4cf2b9a5a9e018275ec1634f9a980d7226e4fd240fa8313c21aaadR27). I remember mentioning in one of your PRs cypress testing isn't testing the database layer. I think we just need to know upon changing the name, the controller redirects to the index page so I think I can get rid of this line.